cli/src/main/java/org/jboss/as/cli/CommandCompleter.java

```diff
@@ -139,6 +142,31 @@ protected int doComplete(CommandContext ctx, String buffer, int cursor, List<Str
                 candidates.remove(notIndex);
             }
         }
+
+        // https://issues.jboss.org/browse/WFCORE-1714
+        if (unresolvedVariableException == false) {
+            int index = buffer.lastIndexOf('$');
+            if (index != -1) {
+                String variable = buffer.substring(index + 1);
+                if (buffer.endsWith(variable)) {
+                    List<String> candidateVariables = new ArrayList<>();
+                    for (String var : ctx.getVariables()) {
+                        if (var.startsWith(variable)) {
+                            candidateVariables.add(var);
+                        }
+                    }
+                    if (candidateVariables.size() > 1) {
+                        candidates.addAll(candidateVariables); // more than one candidate variables
+                        Collections.sort(candidates);
+                        if (!candidateVariables.stream().anyMatch(e -> e.equals(variable)))
+                            return buffer.length() - variable.length();
+                    }
+                } else {
+                    // failed to resolve a variable in the middle of the line
```

Comment:
Just a suggestion, perhaps you could revert if and else to avoid a negative if test.